Tests/Integration/AbstractIntegrationTest.php

```diff
@@ -144,12 +146,17 @@ protected function getRefreshToken(string $encryptedPayload): ?RefreshToken
 
     protected function createAuthorizationRequest(?string $credentials, array $body = []): ServerRequestInterface
     {
-        return $this
+        $request = $this
             ->psrFactory
             ->createServerRequest('', '')
-            ->withHeader('Authorization', sprintf('Basic %s', base64_encode($credentials)))
             ->withParsedBody($body)
         ;
+
+        if (null !== $credentials) {
+            return $request->withHeader('Authorization', sprintf('Basic %s', base64_encode($credentials)));
```

Comment:
@X-Coder264 Do we need this return here? Don't see any real benefit, what if more conditions are added in the future?

Reply:
@HypeMC We need it, because the PSR-7 request is immutable and a new request instance is returned on the `withHeader` call.

Reply:
@X-Coder264 Fair enough, but still what if more conditions are added later, what about `$request = $request->withHeader`? Your call, I'm just writing my thoughts.

Reply:
@HypeMC Good point, done.